src/main/java/io/sealights/plugins/sealightsjenkins/BeginAnalysis.java

```diff
@@ -626,25 +607,29 @@ private SeaLightsPluginInfo createSeaLightsPluginInfo(
         String workingDir = ws.getRemote();
         slInfo.setEnabled(true);
 
-        slInfo.setBuildName(getFinalBuildName(build, slInfo, logger));
+        slInfo.setBuildName(getFinalBuildName(build, additionalProps, logger));
 
         if (workspacepath != null && !"".equals(workspacepath))
             slInfo.setWorkspacepath(workspacepath);
         else
             slInfo.setWorkspacepath(workingDir);
 
-        slInfo.setAppName(JenkinsUtils.resolveEnvVarsInString(envVars, appName));
+        slInfo.setAppName(JenkinsUtils.resolveEnvVarsInString(envVars,
+                additionalProps.get("appname") != null ? additionalProps.get("appname").toString() : null));
         slInfo.setModuleName(moduleName);
-        slInfo.setBranchName(JenkinsUtils.resolveEnvVarsInString(envVars, branch));
+        slInfo.setBranchName(JenkinsUtils.resolveEnvVarsInString(envVars,
+                additionalProps.get("branch") != null ? additionalProps.get("branch").toString() : null));
         slInfo.setFilesIncluded(filesIncluded);
         slInfo.setFilesExcluded(filesExcluded);
         slInfo.setRecursive(recursive);
-        slInfo.setPackagesIncluded(packagesIncluded);
-        slInfo.setPackagesExcluded(packagesExcluded);
+        slInfo.setPackagesIncluded(additionalProps.get("packagesincluded") != null ?
+                additionalProps.get("packagesincluded").toString() : null);
+        slInfo.setPackagesExcluded(additionalProps.get("packagesexcluded") != null ?
+                additionalProps.get("packagesexcluded").toString() : null);
         slInfo.setClassLoadersExcluded(classLoadersExcluded);
-        slInfo.setListenerJar(testListenerJar);
+        slInfo.setListenerJar((String) additionalProps.getProperty("testlistenerjar"));
```

Comment:
Seems like this casting ("**(String)**additionalProps.getProperty('x');" is not needed